### ISSUE TITLE
when data folder becomes accessible, the configs are gone

### ISSUE THREAD

**Issue by sigasigasiga** [ppy/osu#28487]
### Type

Game behaviour

### Bug description

1. go to settings, click "change folder location", pick a folder on a removable hard drive
2. close the game
3. remove the hard drive
4. open the game, osu storage error is shown
5. attach the hard drive
6. click "try again" on the error screen
7. BUG: all the configs are gone

### Screenshots or videos

![Screenshot 2024-06-14 232308](https://github.com/ppy/osu/assets/48245894/8123f5e7-74d8-468b-bf98-80c1090ef8e0)

### Version

2024.521.2

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/15842715/compressed-logs.zip)

**Comment by sigasigasiga** [comment]
maybe related to #26027

**Comment by bdach** [comment]
What does "all the configs" mean in this context?

**Comment by sigasigasiga** [comment]
@bdach 
1. login info
2. user settings
3. beatmap settings ("ignore beatmap skin", "background dim", etc)

and maybe something else that i haven't noticed

**Comment by smoogipoo** [comment]
I don't think that "Try again" button can ever work in its current state without a game restart. There would need to be some series of callbacks to refresh the various config managers, available rulesets, realm database, etc.

**Comment by bdach** [comment]
Maybe we should force one then?

**Comment by peppy** [comment]
It's a bit unfortunate considering how close it is to working, but I'm fine with forcing one for now, as long as there's an attached comment explaining what is required to make it not require the restart.
